downstairs/src/lib.rs

```diff
@@ -120,6 +121,184 @@ impl IOop {
     }
 }
 
+/// Read request for a particular extent
+///
+/// `data` should be an uninitializezd borrowed section of the outgoing buffer
+/// (i.e. from a [`RegionReadResponse`]), to reduce memory copies.
+///
+/// Do not derive `Clone` on this type; cloning the object will break the borrow
+/// of `data`, so it's rarely the correct choice.
+#[derive(Debug, Eq, PartialEq)]
+pub(crate) struct ExtentReadRequest {
+    offset: Block,
+    /// This buffer should be uninitialized; read size is set by its capacity
+    data: BytesMut,
+}
+
+/// Ordered list of read requests, as `(extent, block, count)` tuples
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) struct RegionReadRequest(Vec<(u64, Block, NonZeroUsize)>);
```

Comment:
Hah, we have similar taste – I don't love the tuples, but basically got hung up on naming 😬 

Semantically, a `RegionReadRequest` is the thing which contains a `Vec<..>`, but I was getting stuck on what to call the things in the `Vec`.  They can't be called an `ExtentReadRequest`, because that already the type which is sent to the `Extent`.

Maybe `RegionReadRequest(Vec<RegionReadReq>)`?

Reply:
Okay, I tested this out and it feels fine to me, so I switched from tuples to `RegionReadReq`.